File: src/main/java/io/github/dqualizer/dqlang/archive/loadtesttranslator/modeling/ModeledLoadTest.java
package io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.rqa.definition.Artifact;
import io.github.dqualizer.dqlang.types.rqa.definition.Parametrization;
import io.github.dqualizer.dqlang.types.rqa.definition.ResponseMeasure;
import lombok.Getter;
import lombok.ToString;

import java.util.LinkedHashSet;

@Getter
@ToString
public class ModeledLoadTest {

    private Artifact artifact;
    private String description;
    private Stimulus stimulus;
    private Parametrization parametrization;
    @JsonProperty("response_measure")
    private ResponseMeasure responseMeasure;
    @JsonProperty("result_metrics")
    private LinkedHashSet<String> resultMetrics;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/archive/loadtesttranslator/dqlang/modeling/ModeledLoadTest.kt
package io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Getter
import lombok.ToString

@Getter
@ToString
class ModeledLoadTest {
    val artifact: io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling.Artifact? = null
    val description: String? = null
    val stimulus: io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling.Stimulus? = null
    val parametrization: io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling.Parametrization? = null

    @JsonProperty("response_measure")
    val responseMeasure: io.github.dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling.ResponseMeasure? = null

    @JsonProperty("result_metrics")
    val resultMetrics: LinkedHashSet<String>? = null
}
